wasm/wat/lexer.go

```diff
@@ -14,10 +14,10 @@ import (
 // Returning an error will short-circuit any future invocations.
 type parseToken func(source []byte, tok tokenType, beginLine, beginCol, beginPos, endPos int) error
 
-// lex invokes the parser function for each token, the source is exhausted.
+// lex invokes the parser function for each token, the source is exhausted. The fileName is used in error messages.
 //
 // Errors from the parser or during tokenization exit early, such as dangling block comments or unexpected characters.
-func lex(source []byte, parser parseToken) error {
+func lex(fileName string, source []byte, parser parseToken) error {
```

Comment:
oh I also added the fileName even though we don't use it, yet. I noticed wabt does and it makes sense to.